### PR TITLE
add function to get top models

### LINKED ISSUE #51: Function to display top N models from a randomsearch

### PR DESCRIPTION
**Pull request by farrandi**
- Have not changes the notebook file, will do it in a separate PR after resolving other merge conflicts
- Made the function and testcases

### CODE REVIEW

**Comment by salva-u** [comment]
looks good

**Review by salva-u — APPROVED** [review]

File: tests/test_search_top_models.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        result = fit_and_return_top_models(
+            search, 3, self.X_train, self.y_train, ["param_alpha"], True
+        ).T
+        self.assertEqual("rank_test_score", result.index.name)
+        for column in columns_expected:
+            self.assertIn(column, result.columns, f"Missing expected column: {column}")
+
+    def test_train_score_false(self):
+        """
+        Test that the returned DataFrame has the correct columns
```
Comment (rbouwer) [inline tests/test_search_top_models.py]: This seems to maybe have the wrong docstring

**Comment by rbouwer** [comment]
Added one comment in test file - otherwise LGTM!

**Comment by rbouwer** [comment]
Looks great!

**Review by rbouwer — APPROVED** [review]

**Review by atabak-alishiri — APPROVED** [review]

**Pull request merged**